packages/agreement-process/test/submitAgreement.test.ts

```diff
@@ -0,0 +1,1742 @@
+    }
+
+    expect(selfcareV2ClientMock.getUserInfoUsingGET).toHaveBeenCalled();
+    expect(actualAgreement).toMatchObject({
+      id: submittedAgreement.id,
+      eserviceId: eservice.id,
+      descriptorId: descriptor.id,
+      producerId: producer.id,
+      consumerId: consumer.id,
+      state: toAgreementStateV2(agreementState.active),
+      contract: {
+        id: expect.any(String),
+        name: expect.any(String),
+        prettyName: expect.any(String),
+        contentType: expect.any(String),
+        path: expect.any(String),
+        createdAt: expect.any(BigInt),
+      },
+      consumerNotes: consumerNotesText,
+      verifiedAttributes: [
+        {
+          id: validVerifiedTenantAttribute.id,
+        },
+      ],
+      suspendedByConsumer: false,
+      suspendedByProducer: false,
+      suspendedByPlatform: false,
+    });
+    expect(actualAgreement.stamps?.activation).toMatchObject({
+      who: authData.userId,
+      when: expect.any(BigInt),
+    });
+  });
+
+  it("should submit agreement with new state PENDING when producer is different from consumer and no related agreements exist, generates AgreementSubmitted event", async () => {
+    const consumerId = generateId<TenantId>();
+    const producer = getMockTenant();
+    const consumerNotesText = "This is a test";
+
+    const validVerifiedTenantAttribute: TenantAttribute = {
+      ...getMockVerifiedTenantAttribute(),
+      verifiedBy: [
+        {
+          id: consumerId,
+          verificationDate: new Date(new Date().getFullYear() - 1),
+          expirationDate: new Date(new Date().getFullYear() + 1),
+          extensionDate: undefined,
+        },
+      ],
+    };
+
+    const certifiedTenantAttribute: TenantAttribute = {
+      ...getMockCertifiedTenantAttribute(),
+      revocationTimestamp: undefined,
+    };
+    const declareTenantAttributeMissingInDescriptor: TenantAttribute =
+      getMockCertifiedTenantAttribute();
+
+    const consumer = {
+      ...getMockTenant(consumerId, [
+        validVerifiedTenantAttribute,
+        certifiedTenantAttribute,
+        declareTenantAttributeMissingInDescriptor,
+      ]),
+      selfcareId: generateId<SelfcareId>(),
+      mails: [
+        {
+          id: generateId(),
+          kind: tenantMailKind.ContactEmail,
+          address: "avalidemailaddressfortenant@testingagreement.com",
+          createdAt: new Date(),
+        },
+      ],
+    };
+
+    const verifiedDescriptorAttribute = getMockEServiceAttribute(
+      validVerifiedTenantAttribute.id
+    );
+    const certifiedDescriptorAttribute = getMockEServiceAttribute(
+      certifiedTenantAttribute.id
+    );
+
+    const descriptor = {
+      ...getMockDescriptor(),
+      state: descriptorState.suspended,
+      agreementApprovalPolicy: agreementApprovalPolicy.manual,
+      attributes: {
+        certified: [[certifiedDescriptorAttribute]],
+        declared: [],
+        verified: [[verifiedDescriptorAttribute]],
```

Comment:
Also here I'm not sure why this is not failing, but and agreement becomes pending if it has certified and declare attributes satisfied, so declared should be populated (while verified attributes are not strictly needed here, they are checked only for activation)